src/Compilers/CSharp/Portable/Binder/ExpressionVariableFinder.cs

```diff
@@ -456,7 +456,7 @@ protected override LocalSymbol MakeDeclarationExpressionVariable(DeclarationExpr
                             nodeBinder: _enclosingBinder,
                             typeSyntax: node.Type,
                             identifierToken: designation.Identifier,
-                            kind: LocalDeclarationKind.RegularVariable,
+                            kind: LocalDeclarationKind.DeclarationExpression,
```

Comment:
That applies to pattern variables as well.

---
In reply to: [102559169](https://github.com/dotnet/roslyn/pull/17206#discussion_r102559169) [](ancestors = 102559169)